README.md

```diff
@@ -16,3 +16,86 @@ The project and Python dependencies can be installed from source with:
 ```
 pip3 install .
 ```
+
+## Configuration System
+
+This project provides a configuration system that supplies options (e.g., user name, password) to a command-line interface ([CLI](https://en.wikipedia.org/wiki/Command-line_interface)) tool.
+The configuration system follows a tiered order, allowing options to be specified and overridden from both files and command-line options.
+
+### Configuration Sources
+
+In addition to CLI options, the configuration system loads options from [JSON](https://en.wikipedia.org/wiki/JSON) files located across multiple directories.
+By default, configuration files are named `edq-config.json`.
+This value is customizable, but this document will assume the default is used.
+
+For example, a configuration file containing the `user` and `pass` options might look like this:
+```
+{
+    "user": "edq-user",
+    "pass": "1234567890"
+}
+```
+
+The table below lists the configuration sources in the order they are evaluated.
+All sources are processed in the order shown in the table. 
+The value from the later source in the table overrides the earlier one if there are multiple sources.
+
+| Source   | Description |
+| :-----:  | :---------- |
+| Global   | The global configuration file path is a platform-specific user location by default. |
+| Local    | Local configuration is loaded from the first matching file found, starting in the current directory and moving up to ancestor directories until root. |
+| CLI File | Files passed with `--config-file` are loaded in order, with later files overriding earlier ones. |
+| CLI      | Command-line options override all other configuration sources. |
```

Comment:
What each of these descriptions are describing are slightly different
(and they should all be conceptually the same).
For example, `Local` talks about where the files are located,
but `CLI` just says that it overrides all other sources;
and `CLI Files` mentions the flag, but `CLI` does not.

These should all be consistent.
Since there are more complete sections below,
you may want to mention that this table is a summary and give higher-level descriptions here.